src/legacy/core_plugins/console/np_ready/public/application/containers/editor/legacy/console_history/console_history.tsx

```diff
@@ -0,0 +1,207 @@
+          <ul
+            ref={listRef}
+            onKeyDown={(ev: React.KeyboardEvent) => {
+              if (ev.keyCode === keyCodes.ENTER) {
+                restore();
+                return;
+              }
+
+              let currentIdx = selectedIndex;
+
+              if (ev.keyCode === keyCodes.UP) {
+                ev.preventDefault();
+                --currentIdx;
+              } else if (ev.keyCode === keyCodes.DOWN) {
+                ev.preventDefault();
+                ++currentIdx;
+              }
+
+              const nextSelectedIndex = Math.min(Math.max(0, currentIdx), reqs.length - 1);
+
+              setViewingReq(reqs[nextSelectedIndex]);
+              selectedReq.current = reqs[nextSelectedIndex];
+              setSelectedIndex(nextSelectedIndex);
+              scrollIntoView(nextSelectedIndex);
+            }}
+            role="listbox"
+            className="list-group conHistory__reqs"
+            tabIndex={0}
+            aria-activedescendant={`${CHILD_ELEMENT_PREFIX}${selectedIndex}`}
+            aria-label={i18n.translate('console.historyPage.requestListAriaLabel', {
+              defaultMessage: 'History of sent requests',
+            })}
+          >
+            {reqs.map((req, idx) => {
+              const reqDescription = describeReq(req);
+              const isSelected = viewingReq === req;
+              return (
+                // Ignore a11y issues on li's
+                // eslint-disable-next-line
+                <li
+                  key={idx}
+                  id={`${CHILD_ELEMENT_PREFIX}${idx}`}
+                  className={`list-group-item conHistory__req ${
+                    isSelected ? 'conHistory__req-selected' : ''
+                  }`}
+                  onClick={() => {
+                    setViewingReq(req);
+                    selectedReq.current = req;
+                    setSelectedIndex(idx);
+                    scrollIntoView(idx);
+                  }}
+                  role="option"
+                  onMouseEnter={() => setViewingReq(req)}
+                  onMouseLeave={() => setViewingReq(selectedReq.current)}
+                  onDoubleClick={() => restore(req)}
+                  aria-label={i18n.translate('console.historyPage.itemOfRequestListAriaLabel', {
+                    defaultMessage: 'Request: {historyItem}',
+                    values: { historyItem: reqDescription },
+                  })}
+                  aria-selected={isSelected}
+                >
+                  {reqDescription}
+                  <span className="conHistory__reqIcon">
+                    <i className="fa fa-chevron-right" />
+                  </span>
+                </li>
+              );
+            })}
+          </ul>
+
+          <div className="conHistory__body__spacer" />
+
+          <HistoryViewer settings={settings} req={viewingReq} ResizeChecker={ResizeChecker} />
+        </div>
+
+        <div className="conHistory__footer">
```

Comment:
We can use EUI here too.

```jsx
        <EuiSpacer size="s" />

        <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty color="danger" onClick={() => clear()}>
              {i18n.translate('console.historyPage.clearHistoryButtonLabel', {
                defaultMessage: 'Clear',
              })}
            </EuiButtonEmpty>
          </EuiFlexItem>

          <EuiFlexItem grow={false}>
            <EuiFlexGroup justifyContent="flexEnd" alignItems="center">
              <EuiFlexItem grow={false}>
                <EuiButtonEmpty color="primary" onClick={() => close()}>
                  {i18n.translate('console.historyPage.closehistoryButtonLabel', {
                    defaultMessage: 'Close',
                  })}
                </EuiButtonEmpty>
              </EuiFlexItem>

              <EuiFlexItem grow={false}>
                <EuiButton
                  color="primary"
                  disabled={!selectedReq}
                  onClick={() => restore()}
                >
                  {i18n.translate('console.historyPage.applyHistoryButtonLabel', {
                    defaultMessage: 'Apply',
                  })}
                </EuiButton>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
        </EuiFlexGroup>
```

![image](https://user-images.githubusercontent.com/1238659/64299437-0ae90780-cf2e-11e9-81b8-c9bbd14a1b1b.png)